Fix flaky team discussion test

andy-armstrong opened the pull request:
## [TNL-3247](https://openedx.atlassian.net/browse/TNL-3247)

This re-enables the flaky team discussion test and then tries a few things:
 - removes synchronous AJAX calls from discussion posting as this is deprecated functionality which can't be tested with sinon
 - updates the profile view tests to attach the view to the DOM before testing, as some of the discussion code expects itself to be attached. This was throwing errors in sinon that could have led to problems
 - removes the one-time JQuery initialization that was failing in a flaky way, and replaced it with simple function calls to achieve the same functionality (showing and hiding loading indicators)

### Sandbox
- [x] http://andy-armstrong.sandbox.edx.org/

### Testing
- [x] Manual testing

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @cahrens 
- [x] Code review: @efischer19 

FYI: @jzoldak @benpatterson 

### Post-review
- [x] Squash commits

andy-armstrong commented:
@cahrens @jzoldak I've made some changes here that might help with the flaky tests, but I can't be sure. It at least removes the errors that I was seeing in the console when debugging the tests. Do we have any idea how to run this enough times to feel confident? Should we just do a run all command over and over?

cahrens commented:
You can run just the jasmine tests over and over.

andy-armstrong commented:
jenkins run js (1000 times, please)

andy-armstrong commented:
jenkins run bokchoy

andy-armstrong commented:
@cahrens @jzoldak @efischer19 Well, it only took two test runs to hit the standard five errors, so clearly what I have so far isn't helping. I'll have to investigate further when I get back on Tuesday.

andy-armstrong commented:
jenkins run all

andy-armstrong commented:
@jzoldak @efischer19 @cahrens - Well, the tests failed again even with the new Sinon and the changes in this PR:

https://build.testeng.edx.org/job/edx-platform-accessibility-pr/4889/

andy-armstrong commented:
@benpatterson FYI this is the PR where I've been trying to resolve the flaky tests but with little luck so far.

andy-armstrong commented:
jenkins run js

andy-armstrong commented:
jenkins run js

andy-armstrong commented:
@cahrens @efischer19 @jzoldak I believe that this code is ready for review now. I've removed the one-time JQuery initialization that was failing in a flaky way, and replaced it with simple function calls to achieve the same functionality (showing and hiding loading indicators). I've also removed the last remaining ``async: false`` parameters in the forums JavaScript. I'm going to re-run the JS tests multiple times before merging, but it would be good to have this reviewed.

FYI @benpatterson

andy-armstrong commented:
jenkins run js

andy-armstrong commented:
jenkins run js

andy-armstrong commented:
jenkins run js

andy-armstrong commented on common/static/coffee/src/discussion/utils.coffee:
```diff
@@ -1,16 +1,3 @@
-$ ->
-  if !window.$$contents
```
Reviewers: note that I removed this ``$$contents`` variable after consultation with @jimabramson. It was a legacy piece of code that hasn't been used in a while. The remainder of this clause was converted into methods directly on ``DiscussionUtil`` below.

andy-armstrong commented:
jenkins run js

andy-armstrong commented:
jenkins run js

andy-armstrong commented:
jenkins run js

andy-armstrong commented:
Reviewers: I've had nine clean Jasmine test runs since making my last change (two on pushing up changes and seven manually). I've kicked off a tenth now, but I'm feeling good about these changes. The ``params.$loading.loading is not a function`` (TNL-3247) failure has definitely gone as I've removed that logic entirely. There is a risk that TNL-3007 will still be flaky as I'm re-enabling those tests, but hopefully given ten test runs it should be relatively low frequency if it happens at all.

efischer19 commented:
Looks good to me, :+1:

I think at this point it's about ready to merge into master, but we'll still want to keep an eye out for failures on the tnl-3007 tests for a while. I'll send you the splunk query I had been using before to monitor.

efischer19 commented:
https://build.testeng.edx.org/job/edx-platform-js-pr/5715/

![image](https://cloud.githubusercontent.com/assets/7373924/10341100/505888ee-6ce1-11e5-93c2-fbf4f0271f87.png)

cahrens commented:
:+1: after you disable that other test again. :(

andy-armstrong commented:
Well, on the tenth try it refailed the discussion tests. At this point I'm out of ideas for handling that, so I'll just re-disable it.

andy-armstrong commented:
I've removed the enabling of the other flaky test, and updated the squashed commit and PR to use the correct ticket number. If I get a clean build on this I'll merge as I have two thumbs.
